src/Raven.Server/Documents/Includes/IncludeCountersCommand.cs

```diff
@@ -10,23 +12,54 @@ public class IncludeCountersCommand
     {
         private readonly DocumentDatabase _database;
         private readonly DocumentsOperationContext _context;
-        private readonly StringValues _counters;
+        private readonly Dictionary<string, string[]> _countersBySourcePath;
 
+        public Dictionary<string, string[]> CountersToGetByDocId { get; }
         public Dictionary<string, List<CounterDetail>> Results { get; }
 
-        public IncludeCountersCommand(DocumentDatabase database, DocumentsOperationContext context, StringValues counters)
+        public IncludeCountersCommand(DocumentDatabase database, DocumentsOperationContext context)
         {
             _database = database;
             _context = context;
-            _counters = counters;
-            Results = new Dictionary<string, List<CounterDetail>>();
+
+            CountersToGetByDocId = new Dictionary<string, string[]>(StringComparer.OrdinalIgnoreCase);
+            Results = new Dictionary<string, List<CounterDetail>>(StringComparer.OrdinalIgnoreCase);
         }
 
-        public void Fill(string docId)
+        public IncludeCountersCommand(DocumentDatabase database, DocumentsOperationContext context, string[] counters) 
+            : this(database, context)
         {
-            var details = CountersHandler.GetInternal(_database, _context, _counters, docId, false);
-            Results.Add(docId, details.Counters);
+            _countersBySourcePath = new Dictionary<string, string[]>(StringComparer.OrdinalIgnoreCase)
+            {
+                [string.Empty] = counters
+            };
         }
 
+        public IncludeCountersCommand(DocumentDatabase database, DocumentsOperationContext context, Dictionary<string, HashSet<string>> countersBySourcePath)
+            : this(database, context)
+        {
+            _countersBySourcePath = countersBySourcePath.ToDictionary(kvp => kvp.Key, kvp => kvp.Value.ToArray());
+        }
+
+        public void Fill(Document document)
+        {
+            var docId = document.Id;
+
+            foreach (var kvp in _countersBySourcePath)
```

Comment:
What is that? I think we do not support this in API?

Reply:
yes, there is no support for this in the client api (yet ?)

Reply:
client support for this will be in separate PR